arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx

```diff
@@ -0,0 +1,155 @@
+---
+title: "How to customize your Orbit chain's precompiles"
+sidebar_label: "Customize your chain's precompiles"
+description: "Learn how (and when) to customize your Orbit chain's precompiles"
+author: jasonwan
+sidebar_position: 2
+---
+
+import PublicPreviewBannerPartial from '../partials/_orbit-public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+import UnderConstructionPartial from '../../partials/_under-construction-banner-partial.md';
+
+<UnderConstructionPartial />
+
+There are two ways to customize your chain's precompiles:
+
+1.  Add new methods to to an existing [precompile][precompile_dir_link].
+2.  Create a new precompile.
+
+### Prerequisites
+
+Clone the Nitro repository before you begin:
+
+```shell
+git clone --branch @orbitVersionTag@ <https://github.com/OffchainLabs/nitro.git>
+cd nitro
+git submodule update --init --recursive --force
+```
+
+## Option 1: Add new methods to an existing precompile
+
+Using your favorite code editor, open an existing precompile from the [precompiles/][precompile_dir_link] directory. We'll use `ArbSys.sol` as an example. Open the the corresponding `go` file (`ArbSys.go`) and add a simple `SayHi` method:
+
+```go
+func (con *ArbSys) SayHi(c ctx, evm mech) (string, error) {
+	return "hi", nil
+}
+```
+
+Then, open the corresponding Solidity interface file (`ArbSys.sol`) from the [contracts/src/precompiles/][precompile_dir_link] directory and add the required interface. Ensure that the method name on the interface matches the name of the function you introduced in the previous step, `camelCased`:
+
+```solidity
+function sayHi() external view returns(string memory);
+```
+
+Next, build Nitro by following steps 3-7 of the instructions in [How to build Nitro locally](/node-running/how-tos/build-nitro-locally). Note that if you've already built the Docker image, you still need run the last step to rebuild.
+
+Run Nitro by following the instructions in [How to run a full node](/node-running/how-tos/running-a-full-node#putting-it-all-together).
+
+Once your node is running, you can call `ArbSys.sol` either directly using `curl`, or through Foundry's `cast call`.
+
+### Call your function directly using `curl`
+
+```shell
+curl Your_IP_Address:8547\
+-X POST \
+-H "Content-Type: application/json" \
+--data '{"method":"eth_call","params":[{"from":null,"to":"0x0000000000000000000000000000000000000064","data":"0x0c49c36c"}, "latest"],"id":1,"jsonrpc":"2.0"}'
+```
+
+You should see something like this:
+
+```
+{"jsonrpc":"2.0","id":1,"result":"0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026869000000000000000000000000000000000000000000000000000000000000"}
+```
+
+Which the `0x6869` is the hexify utf8 decode of `hi`
```

Comment:
```suggestion
`0x6869` is the hex-encoded utf8 representation of `hi`
```

Calling @Jason-Wanxt @TucksonDev for a sanity check on this edit 🙏 

Future optimization -> It's not clear to me what this is referring to - I'd expect to see `0x6869` in the above sample output. I do see `...00000002686900000000...`.